libraryBackend/libraryApp/models.py

```diff
@@ -156,22 +156,32 @@ def __str__(self):
 
 
 class Book(models.Model):
-    isbn = models.CharField(max_length=13)
+    isbn = models.CharField(max_length=13, unique=True)
     name = models.CharField(max_length=30)
     publisher = models.CharField(max_length=30)
     author = models.CharField(max_length=30)
     publish_year = models.CharField(max_length=4)
-    quantity = models.IntegerField()
     demand = models.BigIntegerField()
-    is_available = models.BooleanField(default=True)
 
     def __str__(self):
-        return self.name + "_" + self.author
+        return self.name + "_" + self.author + "_" + str(self.id)
+
+    @property
+    def quantity(self):
+        return self.copies.objects.filter(is_available=True).count()
```

Comment:
Quantity should give quantity should give quantity of all the books. If you want no. of available books you could create a different property for the same